include/circt/Dialect/FIRRTL/FIRRTLAttributes.td

```diff
@@ -232,20 +232,34 @@ def InnerSymAttr : AttrDef<FIRRTLDialect, "InnerSym"> {
   let parameters = (ins ArrayRefParameter<"InnerSymPropertiesAttr">:$props);
   let builders = [
     AttrBuilderWithInferredContext<(ins "::mlir::StringAttr":$sym),[{
-      return get(sym.getContext(), 
+      return get(sym.getContext(),
       {InnerSymPropertiesAttr::get(sym.getContext(), sym, 0,
                         StringAttr::get(sym.getContext(), "public"))});
+    }]>,
+    // Create an empty array, represents an invalid InnerSym.
+    AttrBuilder<(ins),[{
+      return get($_ctxt, {});
```

Comment:
True, right now only ports have to use this, just because of the way port symbols are stored. We should update the port symbols storage, so that we don't need invalid `InnerSymAttr` for ports with no symbols. Ideally we shouldn't have to rely on this for anything.